File: pset1/greedy.c
#include <cs50.h>
#include <stdio.h>
#include <math.h>

int coinsNeeded(int money);

int main(void)
{
    float owingChangeFloat = 0;
    printf("O hai! ");
    do {
       printf("How much change is owed?\n");
       owingChangeFloat = get_float();

    } while (owingChangeFloat < 0);

    owingChangeFloat *= 100.0;
    int owingChangeInt = (int) round(owingChangeFloat);
    
    printf("%i\n", coinsNeeded(owingChangeInt));
}

int coinsNeeded(int money)
{
    int neededCoins = 0;
    while(money > 0) {
        if(money >= 25) {
            money -= 25;
            neededCoins++;
        } else if (money >= 10) {
            money -= 10;
            neededCoins++;
        } else if (money >= 5) {
            money -= 5;
            neededCoins++;
        } else if (money >= 1) {
            money -= 1;
            neededCoins++;
        } else {
            break;
        }
    }
    
    return neededCoins;
}

File: pset6/greedy.py
import cs50

def main():
    owingChangeFloat = 0
    print("O hai! ", end="")
    while True:
        print("How much change is owed?")
        owingChangeFloat = cs50.get_float();
        if (owingChangeFloat >= 0):
            break
        
    owingChangeFloat *= 100.0
    owingChangeInt = int(round(owingChangeFloat))
    
    print("{}".format(coinsNeeded(owingChangeInt)))
    
def coinsNeeded(money):
    neededCoins = 0
    while(money > 0):
        if(money >= 25):
            money -= 25
            neededCoins += 1
        elif(money >= 10):
            money -= 10
            neededCoins += 1
        elif(money >= 5):
            money -= 5
            neededCoins += 1
        elif(money >= 1):
            money -= 1
            neededCoins += 1
        else:
            break
    return neededCoins
            

if __name__ == "__main__":
    main()